qiskit/algorithms/optimizers/cmaes.py

```diff
@@ -0,0 +1,400 @@
+            std_vector=np.ones((x0.size)),
+            p_sigma=np.zeros(x0.size),
+            p_c=np.zeros(x0.size),
+            sigma=self.sigma_0,
+            _best_func_hist=[],
+            _worst_func_eval=None,
+            _median_func_hist=[],
+        )
+        self._initialize_termination_criteria()
+        # Initialize static variables
+        self.N = self._state.x.size
+        self.lmbda = 4 + int(3 * np.log(self.N))
+
+        self.mu = int(self.lmbda / 2)
+
+        self._initialize_weights()
+        self.mueff = 1.0 / np.sum(self.weights**2)
+
+        self.cc = (4 + self.mueff / self.N) / (self.N + 4 + 2 * self.mueff / self.N)
+        self.cs = (self.mueff + 2) / (self.N + self.mueff + 5)
+        self.c1 = 2 / ((self.N + 1.3) ** 2 + self.mueff)
+        self.cmu = 2 * (self.mueff - 2 + 1.0 / self.mueff) / ((self.N + 2) ** 2 + self.mueff)
+        self.damps = 1 + 2 * max(0, np.sqrt((self.mueff - 1) / (self.N + 1)) - 1) + self.cs
+        self.chiN = self.N**0.5 * (1 - 1 / (4 * self.N) + 1 / (21 * self.N**2))
+
+    def _initialize_termination_criteria(self) -> None:
+        """
+        Sets the default values for the termination criteria if they haven't been
+        specified by the user. Also removes any entries on the dictionary that don't correspond
+        to a termination criteria.
+        """
+        default_term_crit = {
+            "NoEffectAxis": 0,
+            "TolXUp": 1e4,
+            "TolFun": 1e-12,
+            "TolX": self.sigma_0 * 1e-12,
+            "NoEffectCoord": 0,
+            "ConditionCov": 1e14,
+            "EqualFunValues": 0,
+            "Stagnation": 0,#WIP
+        }
+        # Create default choice of termination criteria if None is passed.
+        if self.termination_criteria is None:
+            self.termination_criteria = default_term_crit
+        # Giving default tolerances to the criteria with None as a parameter.
+        for key, value in self.termination_criteria.items():
+            if key not in default_term_crit:
+                self.termination_criteria.pop(key)
+            if value is None:
+                self.termination_criteria[key] = default_term_crit[key]
+
+    def _initialize_weights(self) -> None:
+        """
+        Initializes the weights of the CMA-ES optimizer.
+        """
+        # Default value for weights.
+        if self.weights is None:
+            self.weights = np.log((self.lmbda + 1) / 2) - np.log(np.arange(1, self.mu + 1))
+
+        # Check if weights are in descending order.
+        if np.any(self.weights[:-1] < self.weights[1:]):
+            raise ValueError("Weights must be in descending order.")
+
+        # Check that all weights are positive.
+        if np.any(self.weights < 0):
+            raise ValueError("Weights must be positive.")
+
+        # Normalize weights.
+        self.weights = self.weights / np.sum(self.weights)
+
+    def _NoEffectAxis(self, tol: float) -> bool:
```

Comment:
These names are not really PEP complliant. Rather than mixed case here can they just be all lower case. Then when you call below you just need to do a lower() on the key when forming the method name to call.